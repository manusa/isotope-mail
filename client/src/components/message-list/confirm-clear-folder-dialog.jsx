import React from 'react';
import PropTypes from 'prop-types';
import {translate} from 'react-i18next';
import Dialog from '../dialog/dialog';

export const ConfirmClearFolderDialog = ({t, visible, cancelAction, deleteAction}) => {
  const actions = [
    {label: t('messageList.clearFolderDialog.cancel'), action: cancelAction},
    {label: t('messageList.clearFolderDialog.deleteAll'), action: deleteAction}
  ];
  return (
    <Dialog
      visible={visible}
      title={t('messageList.clearFolderDialog.title')}
      actions={actions}
    >
      <p>{t('messageList.clearFolderDialog.message')}</p>
    </Dialog>
  );
};

ConfirmClearFolderDialog.propTypes = {
  visible: PropTypes.bool.isRequired,
  cancelAction: PropTypes.func.isRequired,
  deleteAction: PropTypes.func.isRequired
};

export default translate()(ConfirmClearFolderDialog);
