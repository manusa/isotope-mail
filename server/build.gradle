buildscript {
	ext {
		springBootVersion = '2.0.4.RELEASE'
	}
	repositories {
		mavenCentral()
	}
	dependencies {
		classpath("org.springframework.boot:spring-boot-gradle-plugin:${springBootVersion}")
	}
}

apply plugin: 'java'
apply plugin: 'eclipse'
apply plugin: 'org.springframework.boot'
apply plugin: 'io.spring.dependency-management'

group = 'com.marcnuri.isotope'
version = '0.0.1-SNAPSHOT'
sourceCompatibility = 1.8

repositories {
	mavenCentral()
}


dependencies {
	compile('org.springframework.boot:spring-boot-starter')
	compile('org.springframework.boot:spring-boot-starter-web')
	compile('org.springframework.boot:spring-boot-starter-hateoas')
	compile('com.sun.mail:javax.mail:1.6.1')
	testCompile('org.springframework.boot:spring-boot-starter-test')
	testCompile('com.marcnuri:spring-common-test:0.0.2')
}
