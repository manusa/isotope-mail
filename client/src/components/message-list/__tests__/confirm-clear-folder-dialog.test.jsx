import React from 'react';
import {shallow} from 'enzyme/build/index';
import {ConfirmClearFolderDialog} from '../confirm-clear-folder-dialog';

describe('ConfirmClearFolderDialog component test suite', () => {
  describe('Snapshot render', () => {
    test('Defaults, Should render ConfirmClearFolderDialog', () => {
      // Given
      const props = {
        t: jest.fn(messageKey => messageKey),
        visible: true,
        cancelAction: jest.fn(),
        deleteAction: jest.fn()
      };
      // When
      const confirmClearFolderDialog = shallow(<ConfirmClearFolderDialog {...props} />);
      // Then
      expect(confirmClearFolderDialog).toMatchSnapshot();
    });
  });
});
