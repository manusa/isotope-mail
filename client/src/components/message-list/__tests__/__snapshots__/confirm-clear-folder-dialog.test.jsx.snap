// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmClearFolderDialog component test suite Snapshot render Defaults, Should render ConfirmClearFolderDialog 1`] = `
<Dialog
  actions={
    Array [
      Object {
        "action": [MockFunction],
        "label": "messageList.clearFolderDialog.cancel",
      },
      Object {
        "action": [MockFunction],
        "label": "messageList.clearFolderDialog.deleteAll",
      },
    ]
  }
  className=""
  containerClassName=""
  contentClassName=""
  scrimClick={[Function]}
  title="messageList.clearFolderDialog.title"
  visible={true}
>
  <p>
    messageList.clearFolderDialog.message
  </p>
</Dialog>
`;
